mods/mc_worldmanager/hooks.lua

```diff
@@ -1,3 +1,16 @@
+local function makePlayerImmortal(player)
+    local groups = player:get_armor_groups() or {}
+    groups["immortal"] = 1
+    player:set_armor_groups(groups)
```

Comment:
Making a player immortal doesn't reset their HP, so if they took damage, their health bar will still show up after becoming immortal again
Healing the player could be a good way to show the player that they are immortal again, since the health bar is hidden if a player is immortal and at full health

```suggestion
    player:set_armor_groups(groups)
    
    local prop_table = player:get_properties()
    if prop_table.hp_max then
        player:set_hp(prop_table.hp_max, "immortality")
    end
```

Reply:
I like this suggestion, good call!